pkg/karmadactl/cmdinit/bootstraptoken/clusterinfo/clusterinfo_test.go

```diff
@@ -0,0 +1,208 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterinfo
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/kubernetes"
+	clientset "k8s.io/client-go/kubernetes"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+
+	"github.com/karmada-io/karmada/operator/pkg/certs"
+)
+
+func TestCreateBootstrapConfigMapIfNotExists(t *testing.T) {
+	kubeAdminConfig := `
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: %s
+    server: https://test-cluster:6443
+  name: test-cluster
+contexts:
+- context:
+    cluster: test-cluster
+    user: test-user
+  name: test-context
+current-context: test-context
+kind: Config
+preferences: {}
+users:
+- name: test-user
+  user:
+    client-certificate-data: %s
+    client-key-data: %s
+`
+	tests := []struct {
+		name    string
+		client  clientset.Interface
+		cfgFile string
+		prep    func(cfgFile string) error
+		verify  func(clientset.Interface) error
+		cleanup func(cfgFile string) error
+		wantErr bool
+		errMsg  string
+	}{
+		{
+			name:    "CreateBootstrapConfigMapIfNotExists_NonExistentConfigFile_FailedToLoadAdminKubeConfig",
+			prep:    func(string) error { return nil },
+			verify:  func(clientset.Interface) error { return nil },
+			cleanup: func(string) error { return nil },
+			wantErr: true,
+			errMsg:  "failed to load admin kubeconfig",
+		},
+		{
+			name:    "CreateBootstrapConfigMapIfNotExists_WithConfigFile_ConfigMapCreatedInKubePublicNamespace",
+			client:  fakeclientset.NewSimpleClientset(),
+			cfgFile: filepath.Join(os.TempDir(), "config-temp.txt"),
+			prep: func(cfgFile string) error {
+				caKarmadaCert, err := certs.NewCertificateAuthority(certs.KarmadaCertAdmin())
+				if err != nil {
+					t.Fatalf("NewCertificateAuthority() returned an error: %v", err)
+				}
+
+				kubeAdminConfig = fmt.Sprintf(
+					kubeAdminConfig,
+					base64.StdEncoding.EncodeToString(caKarmadaCert.CertData()),
+					base64.StdEncoding.EncodeToString(caKarmadaCert.CertData()),
+					base64.StdEncoding.EncodeToString(caKarmadaCert.KeyData()),
+				)
+
+				err = os.WriteFile(cfgFile, []byte(kubeAdminConfig), 0600)
+				if err != nil {
+					return fmt.Errorf("failed to write kubeAdminConfig to file, got error: %v", err)
+				}
+
+				return nil
+			},
+			cleanup: func(cfgFile string) error {
+				if err := os.Remove(cfgFile); err != nil {
+					return fmt.Errorf("failed to remove config file %s, got an error: %v", cfgFile, err)
+				}
+				return nil
+			},
+			verify: func(c clientset.Interface) error {
+				return verifyKubeAdminKubeConfig(c)
+			},
+			wantErr: false,
```

Comment:
The current test does not cover `Update ConfigMap if it exist`, you can add a TODO item, easy to continue to evolve at a later stage